### PR TITLE
analytics(explore): Adding explore query analytics

### PR DESCRIPTION
**Pull request by Abdkhan14**

### CODE REVIEW

File: static/app/views/explore/hooks/useAnalytics.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+export function useAnalytics({
+  result,
+  resultsMode,
+  visualizes,
+  organization,
+  groupBys,
+  userQuery,
+}: {
+  groupBys: string[];
```
Comment (Zylphrex) [inline static/app/views/explore/hooks/useAnalytics.tsx:17]: Perhaps we should call this `columns` and pass the group bys in aggregate mode and fields in samples mode.

File: static/app/views/explore/hooks/useAnalytics.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    return {
+      query_status: result?.status,
+      results_mode: resultsMode,
+      has_results: Array.isArray(result?.data) && result.data.length > 0,
+      user_query: userQuery,
+      visualizes,
+      organization,
+      groupBys,
+    };
+  }, [result, visualizes, organization, groupBys, userQuery, resultsMode]);
```
Comment (Zylphrex) [inline static/app/views/explore/hooks/useAnalytics.tsx:46]: Can't you put this directly inside `useEffect` and skip an extra `useMemo`?

File: static/app/views/explore/hooks/useAnalytics.tsx (hunk 1 above)
Comment (Abdkhan14) [reply static/app/views/explore/hooks/useAnalytics.tsx]: Yup good idea @Zylphrex

File: static/app/views/explore/hooks/useAnalytics.tsx (hunk 2 above)
Comment (Abdkhan14) [reply static/app/views/explore/hooks/useAnalytics.tsx]: Yup I refactored the code and added counts @Zylphrex

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**